libraries/lib-xml/XMLAttributeValueView.h

```diff
@@ -0,0 +1,174 @@
+   //! Check if view has the StringView type.
+   bool IsStringView() const noexcept;
+
+   //! Try to get a boolean value from the view.
+   bool Get(bool& value) const noexcept;
+   //! Try to get a short value from the view.
+   bool Get(short& value) const noexcept;
+   //! Try to get an unsigned short value from the view.
+   bool Get(unsigned short& value) const noexcept;
+   //! Try to get an int value from the view.
+   bool Get(int& value) const noexcept;
+   //! Try to get an unsigned int value from the view.
+   bool Get(unsigned int& value) const noexcept;
+   //! Try to get a long value from the view.
+   bool Get(long& value) const noexcept;
+   //! Try to get an unsigned long value from the view.
+   bool Get(unsigned long& value) const noexcept;
+   //! Try to get a long long value from the view.
+   bool Get(long long& value) const noexcept;
+   //! Try to get an unsigned long long value from the view.
+   bool Get(unsigned long long& value) const noexcept;
+   //! Try to get a float value from the view.
+   bool Get(float& value) const noexcept;
+   //! Try to get a double value from the view.
+   bool Get(double& value) const noexcept;
+   //! Try to get a string_view value from the view.
+   bool Get(std::string_view& value) const noexcept;
+
+   //! Try to get boolean value or return defaultValue on failure.
+   bool GetBool(bool defaultValue = {}) const noexcept;
+   //! Try to get int16_t value or return defaultValue on failure.
+   int16_t GetInt16(short defaultValue = {}) const noexcept;
+   //! Try to get uint16_t value or return defaultValue on failure.
+   uint16_t GetUInt16(unsigned short defaultValue = {}) const noexcept;
+   //! Try to get int32_t value or return defaultValue on failure.
+   int32_t GetInt32(int32_t defaultValue = {}) const noexcept;
+   //! Try to get uint32_t value or return defaultValue on failure.
+   uint32_t GetUInt32(uint32_t defaultValue = {}) const noexcept;
+   //! Try to get int64_t value or return defaultValue on failure.
+   int64_t GetInt64(int64_t defaultValue = {}) const noexcept;
+   //! Try to get uint64_t value or return defaultValue on failure.
+   uint64_t GetUInt64(uint64_t defaultValue = {}) const noexcept;
+   //! Try to get float value or return defaultValue on failure.
+   float GetFloat(float defaultValue = {}) const noexcept;
+   //! Try to get double value or return defaultValue on failure.
+   double GetDouble(double defaultValue = {}) const noexcept;
+   //! Try to get string_view value or return defaultValue on failure.
+   std::string_view GetStringView(std::string_view defaultValue = {}) const noexcept;
+   //! Convert the view value to an UTF8 string.
+   std::string ToString() const;
+   //! Convert the view value to wxString.
+   wxString ToWString() const;
+
+private:
+   template <typename ResultType>
+   bool GetInteger(ResultType& value) const noexcept;
+   // std::variant requires macOS 10.14, which is significantly higher than
+   // the current target
```

Comment:
As I showed in commit 3acbcb4c2d8ada1e74991be79ecb3844568bcc0a in master, we can use variant, with some cautions.  Use `get_if` not `get`.

Reply:
Well, this will only replace switch by mType to switch by variant::index. I expect the compilation speed to be much worse in the case of a variant though.

Reply:
Who cares, we will be at C++20 with modules speeding up our builds next month.  Right after the similarly easy replatforming.  Yeah.